proxy/nasshp/types.go

```diff
@@ -0,0 +1,51 @@
+package nasshp
+
+import (
+	"net"
+	"strings"
+)
+
+type HostIp string
+
+func (hi HostIp) IsUrl() bool {
+	res := strings.Split(string(hi), ":")
+	if len(res) != 2 {
+		return false
+	}
+	return net.ParseIP(res[0]) == nil
+}
+
+func (hi HostIp) Resolve() []string {
+	res := strings.Split(string(hi), ":")
+	if len(res) != 2 {
+		return []string{}
+	}
+	p := res[1]
+	ips, err := net.LookupHost(res[0])
+	if err != nil {
+		return []string{}
+	}
+	var toReturn []string
+	for _, ip := range ips {
+		toReturn = append(toReturn, strings.Join([]string{ip, ":", p}, ""))
+	}
+	return toReturn
+}
+
+type Verdict int
+
+const (
+	// Can't decide either way. This is useful for chaning filters.
+	VerdictUnknown Verdict = iota
+	// Let the request in.
+	VerdictAllow
+	// Block the request.
+	VerdictDrop
+)
+
+func (v Verdict) MergePreferAllow(vv Verdict) Verdict {
```

Comment:
So, there are different ways to Merge multiple verdicts? This will return allow as long as one verdict is allow, correct? eg, if either verdict is allow, then allow. 

Was that the intent? Or was the intent to allow only if all are Allow or Unknown?

I would rename the function.

Also.. there is no need to test all resolutions, if at the first one we find allowed, it's guaranteed to be allowed?

Reply:
I removed this, it was going to be a precursor to OPA, but it isn't needed